.github/workflows/client-compatibility.yml

```diff
@@ -0,0 +1,39 @@
+name: Tests against Java Driver Verions
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main" ]
+
+jobs:
+  install:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        driver_version: [ 4.10.0, 4.11.0, 4.11.1, 4.11.2, 4.11.3, 4.12.0, 4.12.1, 4.13.0, 4.14.0, 4.15.0, 4.16.0, 4.17.0 ]
```

Comment:
TODO: this is testing against all known releases of datastax/java-driver.  At some point in the future this should be tweaked to enable testing against both datastax/java-driver and apache/cassandra-java-driver.

This can definitely be fixed at some point in the future.